csv-bulk-signature/src/main.ts

```diff
@@ -160,10 +163,30 @@ async function zipUpFile() {
   console.log("👍 Zip file created (signatures + report)");
 }
 
-async function createStatusReport({
-  skippedRows,
-  processedRows,
-}: StatusReport) {
+async function createStatusReport(contacts: Contact[]) {
+  const skippedRows: string[] = [];
+  const processedRows: string[] = [];
+
+  // 1. collect skipped and processed rows
+  contacts.forEach((contact, index) => {
+    const hasRequiredData = checkRequiredFields(contact);
+    if (!hasRequiredData) {
+      skippedRows.push(contact["Full Name*"]);
+    } else {
+      processedRows.push(contact["Full Name*"]);
+    }
+  });
+
+  // 2. report in console
+  console.log(
+    `${skippedRows.length ? "🙅‍♂️" : "👍"} Signatures:`,
+    processedRows.length,
+    `processed and`,
+    skippedRows.length,
+    `skipped`
+  );
+
+  // 3. report in file
```

Comment:
so we can move report consts away from being a global, here we move all reporting into this report function